### ISSUE TITLE
New target features & configurations

### ISSUE THREAD

**Issue by refracta** [refracta/dcss-webtiles-server#95]
- [x] Support bcrawl
- [x] Support [yiufcrawl](https://github.com/yrmvgh/yiufcrawl) 
- [ ] Support Sizzell
- [x] Support experimental build
- [x] cron - ttyrec compressing
- [x] #97
- [x] #96
- [x] wong .rc, .macro, where, lst nginx content format
- [x] Fix dgl database error related to show_save_info
- [x] server_id in config.py
- [x] static known_host
- [x] Check announcements.log
- [x] UPS Maintainence
- [x] Data backup solution - AWS Glacier Deep Archive
- [ ] Bot Integration
- [x] Container safe stop logic (game save loss)
- [x] dgl watch problem
- [x] fooTV 

- [x] NAO - WebTerm
- https://github.com/butlerx/wetty
- https://github.com/tsl0922/ttyd
- [ ] CPO - Weekly score board
- [ ] CUE - Patron API
- [x] CXC - Server Monitor (netdata)
- [x] CDI - Dropdown versions
- [x] CAO - Game slot
- [x] Fix ttyrec compress logic
- [x] rcfinder.html
- [ ] rebuild-trigger permission
- [x] renice option
- [x] gooncrawl explbr
- [x] lock file issue

**Comment by Rytisgit** [comment]
- - [x] in case you want to add checkmarks, you can use `- [ ] sample` to add the checkmark box with markdown 
- Also Webterm in crawl is not really necessary, because you can use an rc option to get the ASCII view, by using `tile_display_mode = glyphs` or `tile_display_mode = hybrid` in rc settings

**Comment by refracta** [comment]
The checkboxes look much cleaner, thank you.

That option is a bit different from the WebTerm I want. The official NetHack server, NAO server, has a [feature that allows playing dgamelaunch sessions on the web](https://alt.org/nethack/hterm/) using WebSocket (maybe). I would like to support this if possible (although it's a very low-priority issue).

**Comment by refracta** [comment]
By the way, there is an issue where the trunk update via cron does not occur automatically. This could be due to the modifications I have made so far, but I would like to know if the cron was working well (tested) in your original project from which I forked.

If the cron was working well in your original project, it is likely that the issue is not with the cron itself, but rather with the DGL script.

**Comment by Rytisgit** [comment]
Yes, the cron was not working well to update the trunk for me either, this is why I mentioned that it is unreliable in the issues of my project, Perhaps it's not cron itself but the way it's running the update commands. But it seems that it has carried over to your project as well.

I do know about the webterm you are speaking about, I just mentioned the options in rc, in case you thought it was terminal-like enough.

**Comment by refracta** [comment]
Thank you for letting me know. Since recognizing the issue two days ago, I have set up a manual process to update every 15 minutes, but this is just a temporary solution. This problem is probably the most urgent one to resolve. 

I have experienced issues where certain environment variables were not correctly applied in the cron environment within a container. Therefore, I had to hardcode paths to make it work. 
This might be a similar problem.
https://gist.github.com/refracta/559e91a9d4686624556a6733c2191cdf (`node` not work, `/usr/local/bin/node` work properly)

**Comment by refracta** [comment]
I confirmed that the cron process functions properly when started manually from within the container shell. Therefore, I added the cron command to the startup script. I need to resolve some build issues currently encountered to proceed with production test, but I believe this problem likely has been fixed.
https://github.com/refracta/dcss-server/commit/eb6ca1272b139e15d8e59551472bd0ff61afa8fe
